components/automate-chef-io/README.md

```diff
@@ -81,11 +88,314 @@ This will create the file and fill in the necessary frontmatter automatically.
 
 Hugo allows us to nest our data directory structure as much as necessary. You can add as many folders as necessary under `components/automate-chef-io/data/docs`.
 
-```
+```output
 .
 ├── data
 │   ├── docs
 │   │   ├── cli
 |   │   │   ├── command_one.yml
 |   │   │   └── command_two.yml
 ```
+
+## API Documentation
+
+To view the API documentation locally, 1. From the `components/automate-chef-io` directory, run the following command to start the development server.
+
+```shell
+$ make serve
+```
+
+To view the API docs in your browser, navigate to:
+
+```shell
+http://localhost:1313/docs/api
+```
+
+### Building API Docs
+
+Chef Automate  generates API documentation from protoc files comments using
+
+* [grpc-gateway protoc-gen-swagger](https://github.com/grpc-ecosystem/grpc-gateway/tree/master/protoc-gen-swagger) as the protoc compiler
+* [OpenAPI 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md) as the API schema
+* [ReDoc](https://github.com/Redocly/redoc) as the front-end display
+
+To build the API documentation during development:
+
+1. Make any edits to the .proto files necessary.
+2. Recompile the .proto files as described in the [Automate development guide](https://github.com/chef/automate/blob/master/dev-docs/DEV_ENVIRONMENT.md)
+3. Sync the .swagger files generated with those in the docs component: `make sync_swagger_files`
+4. Run the hugo server and view the documentation in your browser
+
+Rebuilding the API docs is necessary when adding documentation content to Automate's .proto files. Changing the code in the .proto files does not require rebuilding, but new or updated services, messages, or fields require documentation.
+Our goal is to provide timely and accurate documentation for Chef Automate users.
+
+You can build the API docs by running the following all-in-one command from the top-level `automate` directory:
+
+`hab studio run "source .studiorc && compile_all_protobuf_components" && pushd components/automate-chef-io/ && make sync_swagger_files generate_swagger && make serve || popd`
+
+Review the API docs in your browser:
+
+`http://localhost:1313/docs/api`
+
+#### Notes on building the docs
+
+* If you end up with merge conflicts while writing API docs, don't attempt to resolve them in any generated files. It's better to resolve the conflicts in any human edited file and then regenerate the docs.
+* The all-in-one build command above can fail somewhere in the middle and leave things in an inconsistent state.
+It is safe to run multiple times in a row while fixing syntax errors in .proto files or similar, but since it changes directories it can sometimes be necessary to change back to the root automate directory.
```

Comment:
```suggestion
It is safe to run this command multiple times in a row while fixing syntax errors in .proto files or similar, but since it changes directories, it can sometimes be necessary to change back to the root automate directory.
```